packages/orchestrator/internal/sandbox/block/cache.go

```diff
@@ -358,76 +358,146 @@ func NewCacheFromProcessMemory(
 func (c *Cache) copyProcessMemory(
 	ctx context.Context,
 	pid int,
-	ranges []Range,
+	rs []Range,
 ) error {
+	splittedRanges, err := splitRanges(rs, MAX_RW_COUNT)
+	if err != nil {
+		return fmt.Errorf("failed to split ranges: %w", err)
+	}
+
 	var start int64
 
-	for i := 0; i < len(ranges); i += IOV_MAX {
-		segmentRanges := ranges[i:min(i+IOV_MAX, len(ranges))]
+	// We need to split the ranges because the Kernel does not suppor reading/writing more than MAX_RW_COUNT bytes in a single operation.
```

Comment:
```suggestion
	// We need to split the ranges because the Kernel does not support reading/writing more than MAX_RW_COUNT bytes in a single operation.
```